src/features/island/buildings/components/building/AddOil.tsx

```diff
@@ -0,0 +1,110 @@
+import {
+  BUILDING_DAILY_OIL_CONSUMPTION,
+  BUILDING_OIL_BOOSTS,
+} from "features/game/events/landExpansion/cook";
+import { BUILDING_DAILY_OIL_CAPACITY } from "features/game/events/landExpansion/supplyCookingOil";
+
+interface Props {
+  buildingId: string;
+  buildingName: CookingBuildingName;
+}
+
+export const AddOil: React.FC<Props> = ({ buildingName, buildingId }) => {
+  const { gameService } = useContext(Context);
+  const { t } = useAppTranslation();
+
+  const [showOilModal, setShowOilModal] = useState(false);
+
+  const selectOil = (state: MachineState) =>
+    state.context.state.buildings?.[buildingName]?.[0].oilRemaining ?? 0;
+  const selectAvailable = (state: MachineState) =>
+    state.context.state.inventory.Oil ?? new Decimal(0);
+
+  const barrelOil = useSelector(gameService, selectOil);
+  const availableOil = useSelector(gameService, selectAvailable);
+
+  const handleSupplyOil = () => {
+    gameService.send("cookingOil.supplied", {
+      building: buildingName,
+      buildingId,
+      oilQuantity: 1,
+    });
+
+    gameService.send("SAVE");
+  };
+
+  return (
+    <>
+      <div className="p-1">
+        <Label type="default" className="mb-2" icon={barrel}>
+          {t("building.oil.remaining", { oil: barrelOil })}
+        </Label>
+
+        <div className="flex items-center mt-2 -mx-2">
+          <Button className="w-32" onClick={() => setShowOilModal(true)}>
+            {`Add oil`}
+          </Button>
+        </div>
+      </div>
+      <Modal show={showOilModal} onHide={() => setShowOilModal(false)}>
+        <CloseButtonPanel>
+          <div className="p-1">
+            <div className="flex">
+              <Label type="default" className="mb-2 mr-2" icon={barrel}>
+                {t("building.oil.remaining", { oil: barrelOil })}
+              </Label>
+              {barrelOil >= BUILDING_DAILY_OIL_CAPACITY[buildingName] ? (
+                <Label type="warning" className="mb-2">
+                  {`Full tank`}
+                </Label>
+              ) : null}
+            </div>
+
+            <p className="text-sm">
+              {`Boost your cooking in the ${buildingName} with oil! Enjoy a ${
+                BUILDING_OIL_BOOSTS[buildingName] * 100
+              }% cooking time reduction. Note: ${buildingName} uses ${
+                BUILDING_DAILY_OIL_CONSUMPTION[buildingName]
+              } oil daily, adjusted for cooking times.`}
+            </p>
+            <div className="flex justify-between items-center mb-2 mt-4 pl-1">
+              <Label type="default" icon={SUNNYSIDE.icons.basket}>
+                {t("greenhouse.insertOil", { oil: availableOil })}
+              </Label>
+            </div>
+
+            <div className="flex items-center mt-2 -mx-2">
+              <Box count={new Decimal(availableOil)} image={oilIcon} disabled />
+              <Button
+                className="w-12"
+                onClick={() => handleSupplyOil()}
+                disabled={
```

Comment:
<img width="459" alt="Screenshot 2024-05-28 at 3 05 46 PM" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/02c7ec1d-6df3-49e3-8450-ccca8b24c87e">

It looks like the AddOil state is not being reset when it closes.

1. Open Fire Pit
2. Open Add Oil Modal
3. Click "Add Oil" (Modal Closes)
4. Open Add Oil Modal Again
5. The old values are still showing (Oil to add: 2) even though I've maxed out the oil